### DISCUSSION TITLE
Resolve Types in Platform Code

### DISCUSSION THREAD

**Discussion by james1301** [PrismLibrary/Prism#2268] [Xamarin.Forms]
What would be the recommended approach to resolve types in platform specific code? The platform initializer allows you to register implementations from the platform. But how should you do it the other way round?

I am trying to using the PRISM container rather than the service container from this example: [https://docs.microsoft.com/en-us/azure/developer/mobile-apps/notification-hubs-backend-service-xamarin-forms#create-a-cross-platform-xamarinforms-application](url).

So in this example the platform is trying to provide it's information when available and the shared code of using that information is implemented in the shared project.

**Comment by brianlagunas** [comment]
Check out the docs on this topic: https://prismlibrary.com/docs/dependency-injection/platform-specific-services.html

**Reply by james1301** [reply]
Thanks @brianlagunas I have read that and I am using it for a few services. I get how you can register services from the specific platform. But I didn't quite get the best way to resolve a type in the specific platform as the initializer occurs before any other types get registered in my app.xaml.cs.

**Reply by brianlagunas** [reply]
Use the Container property off the Application.

**Comment by dansiegel** [comment]
Better way introduce your own interface and implement that on your PlatformInitializer

```cs
public interface IPlatformStartup
{
    void Start(IContainerProvider container);
}
```

Then in your App you can do something like:

```cs
protected override void OnInitialized()
{
    if(PlatformInitializer is IPlatformStartup ps)
    {
        ps.Start(Container);
    }
}
```

now you have an easy way to manage this from shared code.
